xonsh/built_ins.py

```diff
@@ -720,6 +724,7 @@ def load_builtins(execer=None, config=None, login=False, ctx=None):
     builtins.__xonsh_execer__ = execer
     builtins.__xonsh_commands_cache__ = CommandsCache()
     builtins.__xonsh_all_jobs__ = {}
+    builtins.__xonsh_enabled_builtin_commands__ = set(all_builtin_commands.keys())
```

Comment:
> I guess the problem that you are trying to solve with this is that you want users to be able to load/unload these commands en masse?

Not necessarily en masse, but on demand (same as `enable` works in bash).

I guess my main concern is that it feels like it would be nice to keep user-defined aliases separate from built-in features of the shell.

> (1) it takes away from some of the agency of the users since it seems like they shouldn't touch builtins

I don't agree with this.  Since aliases are loaded first, it is still just as easy to override the behavior of a built-in (just add an alias, as before!).

> (2) it becomes another thing for folks to learn, which makes it harder to teach.

I think the change would be transparent to most users, since aliases still work.  The only interaction users would have to have with it is via `enable`, if necessary.

I guess part of my disagreement here comes from the name "aliases," which seems (to me) to disagree with how other shells use that term.  But I can't come up with a better name.  commands?